### ISSUE TITLE
maybe a rotation bug when do random_flip

### ISSUE THREAD

**Issue by hughlee815** [tianweiy/CenterPoint#136]
hi,

I visualized the waymo data, and i thought your code([this line](https://github.com/tianweiy/CenterPoint/blob/cb25e870b271fe8259e91c5d17dcd429d74abc91/det3d/core/sampler/preprocess.py#L810) and [this line](https://github.com/tianweiy/CenterPoint/blob/cb25e870b271fe8259e91c5d17dcd429d74abc91/det3d/core/sampler/preprocess.py#L827)) may be not so correct.
The waymo coordinate system and heading alpha is defined as the following picture.
![image](https://user-images.githubusercontent.com/58172706/116389137-33ceb680-a84f-11eb-9ceb-17bd7bd89b46.png)
In theory, when we flip the whole scene along x, heading_after = -heading_before; when along y, heading_after = pi - heading_before.
But i am not so sure of this conclusion, so I hope i could check this with you.

**Comment by tianweiy** [comment]
https://github.com/tianweiy/CenterPoint/blob/0d09ec9771509954c30dd653ac7436ff562fd2a8/det3d/datasets/waymo/waymo_common.py#L269

There is an additional conversion here.

**Comment by Super-Tree** [comment]
@tianweiy 
The definition of bbox‘s yaw in waymo open dataset is from positive x axis to positive y axis, so it’s counterclockwise rotation.
https://github.com/tianweiy/CenterPoint/blob/0d09ec9771509954c30dd653ac7436ff562fd2a8/det3d/datasets/waymo/waymo_common.py#L267

**Comment by tianweiy** [comment]
any referenced link for this?

**Comment by Super-Tree** [comment]
@tianweiy  sorry, I can't find the public definition of the bbox's rotation. But I visualized a pair of pointcloud, image and a vehicle bbox:
![image](https://user-images.githubusercontent.com/24618712/116512748-122b0900-a8fb-11eb-8b0f-a88fa87d3b6b.png)

The rotation of this vehicle bbox is 1.2955, 74degree. Meanwhile, the red axis in pointcloud is x axis, green axis is y axis, same as waymo.
p.s. vehicle bbox,xyz lwh,yaw: [57.06236705 -8.04560207  1.03982428  4.47835499  1.93151462  1.57,   1.29551829 ]

**Comment by tianweiy** [comment]
Thanks, I will check this tomorrow.

**Comment by Super-Tree** [comment]
Any update? @tianweiy

**Comment by tianweiy** [comment]
I am too busy with paper deadlines so probably won't time to fully check this until next month. It doesn't seem to affect the performance much as we are already near top(good if this can give us extra boost ...

**Comment by tianweiy** [comment]
@Super-Tree you are right, Waymo is counter clockwise from X to Y. And the coordinate are 

```
                # Waymo: x, y, z, length, width, height, rotation from positive x axis counterclockwisely
                # KITTI: x, y, z, width, length, height, rotation from negative y axis clockwisely 
```
 In addition to this notation, I think the other part of the code are still correct (as all of our rotation matrix formulation assume positive is clockwise

**Comment by tianweiy** [comment]
https://github.com/tianweiy/CenterPoint/blob/400ae24bb4e8f4ccabd46757738bb0304bfa2681/det3d/core/bbox/box_torch_ops.py#L135

**Comment by maudzung** [comment]
I think it's correct. The convention here is different from other repos.
The heading angle is encoded with the formula:

```
gt_boxes[:, -1] = -np.pi / 2 - gt_boxes[:, -1] 
```
